### PR TITLE
[APPC-1249] Implement new auth with backend

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

**NOTE: DON’T ACCEPT THIS UNTIL THE MICROSERVICES TICKET IS DONE**
This ticket adds the ewt (jwt variation) in the services described in the ticket 

**Database changed?**

    No

**Where should the reviewer start?**

EwtAuthorizationInterceptor.kt

**How should this be manually tested?**

Check the logs and check if the service is there

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/APPC-1249

**Questions:**

The first request takes a bit of time to be completed.

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package com.asfoundation.wallet
+
+import com.appcoins.wallet.bdsbilling.WalletService
+import com.asfoundation.wallet.util.convertToBase64
+import com.google.gson.JsonObject
+
+
+private const val TTL = 3600
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt]: Which unit is this variable and what does it represents?

File: app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  fun getNewEwtAuthentication(address: String, currentUnixTime: Long): String {
+    val payload = replaceInvalidCharacters(getPayload(address, currentUnixTime))
+    val signedPayload = walletService.signContent(payload)
+        .blockingGet()
+    return buildAndSaveEwtString(replaceInvalidCharacters(header.convertToBase64()), payload,
+        signedPayload, address,
+        currentUnixTime)
+  }
+
+  private fun shouldBuildEwtAuth(address: String, currentUnixTime: Long): Boolean {
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt]: Should the current timestamp be passed as argument? Can't the class be tricked in this way by receiving the same current timestamp?

File: app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    return cachedAuth[address] == null || hasExpired(currentUnixTime, cachedAuth[address]?.second)
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt]: Please prevent the null propagation very time that it's possible

File: app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      cachedAuth[address]!!.first
+    }
+  }
+
+  @Synchronized
+  fun getNewEwtAuthentication(address: String, currentUnixTime: Long): String {
+    val payload = replaceInvalidCharacters(getPayload(address, currentUnixTime))
+    val signedPayload = walletService.signContent(payload)
+        .blockingGet()
+    return buildAndSaveEwtString(replaceInvalidCharacters(header.convertToBase64()), payload,
```
Comment (JoaoRaimundo) [inline app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt]: If the method is called buildAndSaveEwtString can't we build all the values (payload, signedPayload and header) in that method?

File: app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt (hunk 2 above)
Comment (JDSM01) [reply app/src/main/java/com/asfoundation/wallet/EwtAuthenticatorService.kt]: **Discuss**

**Comment by sama11y** [comment]
Why

**Pull request closed without merging**